modules/user/user.repository.js

```diff
@@ -57,8 +57,31 @@ class UserRepository {
         return userDB;
     }
 
-    async readAllUsers() {
-        const usersArr = await UserModel.find({}, userAggregateForm);
+    async readAllUsers(searchParams, sortParams, limit, skip) {
+        const searchParamsMongo = {};
+        Object.entries(searchParams).map(function ([key, value]) {
+            searchParamsMongo[key] = new RegExp(value, "gi");
+        });
+
+        const sortParamsMongo = {};
+        if (sortParams.sortBy && sortParams.orderBy) {
+            sortParamsMongo[sortParams.sortBy] =
+                sortParams.orderBy === "desc" ? -1 : 1;
+        } else {
+            sortParamsMongo.id = 1;
+        }
+
+        const usersArr = await UserModel.find(
+            searchParamsMongo,
+            userAggregateForm
+        )
+            .limit(limit)
+            .skip(skip)
+            .sort(sortParamsMongo);
+
+        if (!usersArr.length) {
```

Comment:
there shouldn't be any error, just return empty array to client

- [ ] send empty arr to client